kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -228,9 +228,24 @@ class FlinkProcessBuilder(
         hadoopCp.foreach(classpathEntries.add)
         val extraCp = conf.get(ENGINE_FLINK_EXTRA_CLASSPATH)
         extraCp.foreach(classpathEntries.add)
-        if (hadoopCp.isEmpty && extraCp.isEmpty) {
-          warn(s"The conf of ${FLINK_HADOOP_CLASSPATH_KEY} and " +
-            s"${ENGINE_FLINK_EXTRA_CLASSPATH.key} is empty.")
+
+        val hasHadoopJar = Paths.get(flinkHome)
+          .resolve("lib")
+          .toFile
+          .listFiles(new FilenameFilter {
+            override def accept(dir: File, name: String): Boolean = {
+              name.startsWith("hadoop-client") ||
+              name.startsWith("flink-shaded-hadoop")
+            }
+          }).nonEmpty
+
+        if (!hasHadoopJar) {
+          hadoopCp.foreach(classpathEntries.add)
+        }
+
+        if (!hasHadoopJar && hadoopCp.isEmpty && extraCp.isEmpty) {
+          warn(s"No Hadoop jar detected, and the conf of ${FLINK_HADOOP_CLASSPATH_KEY} " +
```

Comment:
No hadoop client jars are found in ${flinkHome}/lib, and ...

update the following KyuubiException message too.

do we need to update docs too.